packages/block-library/src/query-pagination/query-pagination-provider.js

```diff
@@ -0,0 +1,56 @@
+/**
+ * WordPress dependencies
+ */
+import {
+	createContext,
+	useState,
+	useMemo,
+	useContext,
+} from '@wordpress/element';
+import { useSelect } from '@wordpress/data';
+import { store as blockEditorStore } from '@wordpress/block-editor';
+
+const QueryPaginationContext = createContext();
+export default function QueryPaginationProvider( { clientId, children } ) {
+	const paginationArrow = useSelect( ( select ) => {
+		const { getBlocks } = select( blockEditorStore );
+		const innerBlocks = getBlocks( clientId );
+		/**
+		 * Query Pagination Next/Previous blocks' arrows should be synced
+		 * inside a Query Pagination block. So we initialize the context
+		 * arrow value from the the first matching Query Pagination
+		 * Next/Previous block we find.
+		 */
```

Comment:
Why not just use an actual block context (instead of React context) and store the attribute in the "pagination block" instead. You avoid all the sync hustle

Reply:
I followed this approach eventually.